dotnetv3/Cognito/Actions/CognitoWrapper.cs

```diff
@@ -0,0 +1,362 @@
+    /// </summary>
+    /// <param name="cognitoService"></param>
+    public CognitoWrapper(IAmazonCognitoIdentityProvider cognitoService)
+    {
+        _cognitoService = cognitoService;
+    }
+
+    // snippet-start:[Cognito.dotnetv3.ListUserPools]
+    /// <summary>
+    /// List the Amazon Cognito user pools for an account.
+    /// </summary>
+    /// <returns>A list of UserPoolDescriptionType objects.</returns>
+    public async Task<List<UserPoolDescriptionType>> ListUserPoolsAsync()
+    {
+        var response = new ListUserPoolsResponse();
+        var request = new ListUserPoolsRequest { MaxResults = 10 };
+        var userPools = new List<UserPoolDescriptionType>();
+
+        do
+        {
+            response = await _cognitoService.ListUserPoolsAsync(request);
+
+            if (response.UserPools.Count > 0)
+            {
+                userPools.AddRange(response.UserPools);
+            }
+
+            request.NextToken = response.NextToken;
+        }
+        while (response.NextToken is not null);
+
+        return userPools;
+    }
+
+    // snippet-end:[Cognito.dotnetv3.ListUserPools]
+
+    // snippet-start:[Cognito.dotnetv3.ListUsers]
+    /// <summary>
+    /// Get a list of users for the Amazon Cognito user pool.
+    /// </summary>
+    /// <param name="userPoolId">The user pool Id.</param>
+    /// <returns>A list of users.</returns>
+    public async Task<List<UserType>> ListUsersAsync(string userPoolId)
+    {
+        var request = new ListUsersRequest
+        {
+            UserPoolId = userPoolId
+        };
+
+        var response = new ListUsersResponse();
+        var users = new List<UserType>();
+
+        do
+        {
+            response = await _cognitoService.ListUsersAsync(request);
+            users.AddRange(response.Users);
+            request.PaginationToken = response.PaginationToken;
+        }
+        while (response.PaginationToken is not null);
+
+        return users;
+    }
+
+    // snippet-end:[Cognito.dotnetv3.ListUsers]
+
+    // snippet-start:[Cognito.dotnetv3.AdminRespondToAuthChallenge]
+    public async Task<AuthenticationResultType> AdminRespondToAuthChallengeAsync(string userPoolId, string userName, string clientId, string mfaCode, string session)
+    {
+        var challengeResponses = new Dictionary<string, string>();
+        challengeResponses.Add("USERNAME", userName);
+        challengeResponses.Add("SOFTWARE_TOKEN_MFA_CODE", mfaCode);
+
+        var request = new AdminRespondToAuthChallengeRequest
+        {
+            ClientId = clientId,
+            UserPoolId = userPoolId,
+            ChallengeResponses = challengeResponses,
+            Session = session
+        };
+
+        var response = await _cognitoService.AdminRespondToAuthChallengeAsync(request);
+        return response.AuthenticationResult;
+    }
+
+    // snippet-end:[Cognito.dotnetv3.AdminRespondToAuthChallenge]
+
+    // snippet-start:[Cognito.dotnetv3.RespondToAuthChallenge]
```

Comment:
This snippet is not referenced in the metadata, but IS referenced from the scenario. Is that correct according to the spec? Or should it be the Admin version?

Reply:
Add it to the metadata because it was part of the list of actions, but it is the Admin one that should be called from the scenario.